File: Example/OCSlimProject_FitnesseTests/FitnesseTestSuiteXMLResultParserTests.m
#import <XCTest/XCTest.h>
#import "FitnesseTestSuiteXMLResultParser.h"
#import "OCSlimProjectTestDataManager.h"

@interface FitnesseTestSuiteXMLResultParserTests : XCTestCase

@property (nonatomic, strong) FitnesseTestSuiteXMLResultParser *parser;

@end

@implementation FitnesseTestSuiteXMLResultParserTests

- (void)setUp {
    [super setUp];
    
    self.parser = [[FitnesseTestSuiteXMLResultParser alloc] init];
}

- (void)tearDown {
    self.parser = nil;
    
    [super tearDown];
}

- (void) testSuiteJUnitXMLFailureWillFail {
    
    NSData *data = [OCSlimProjectTestDataManager failedResultData];
    
    XCTAssertFalse([self.parser resultForTestSuiteXMLData:data]);
}

- (void) testSuiteJUnitXMLSuccessWillPass {
    
    NSData *data = [OCSlimProjectTestDataManager successResultData];
    
    XCTAssertTrue([self.parser resultForTestSuiteXMLData:data]);
    
}

@end

File: Example/OCSlimProject_FitnesseTests/FitnesseTestSuiteXMLResultParserTests.swift
import XCTest

// Fitnesse Launched as Pre Script in Test Phase
// Test Setup Runslocalhost:8080?Suite&format=junit
// Results Gathered into an XML File
// For Each TestSuite.TestCase element instantiate an XCTestCase

class FitnesseTestSuiteXMLResultParserTests: XCTestCase, XCTestObservation {
    
    let test = FitnesseTestSuiteXMLResultParser()
    let testData = OCSlimProjectTestDataManager()
    
    func testSuiteJUnitXMLFailureWillFail() {
        let data = testData.failedResultData
        XCTAssertFalse(test.resultForTestSuiteXMLData(data))
    }
    
    func testSuiteJUnitXMLSuccessWillPass() {
        let data = testData.successResultData
        XCTAssertTrue(test.resultForTestSuiteXMLData(data))
    }

    
}
